processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/BlacklistedKeysFilter.java

```diff
@@ -37,11 +38,18 @@ public BlacklistedKeysFilter(ProcessorProperties props) {
              .listen((oldValue, newValue) -> ignoreKeys = new HashSet<>(newValue));
     }
 
-    public boolean shouldTake(ConsumerRecord<String, byte[]> record) {
+    public boolean shouldTake(ConsumerRecord<byte[], byte[]> record) {
+        final byte[] key = record.key();
+        if (key == null) {
+            return true;
+        }
+
+        final String stringKey = new String(key, StandardCharsets.UTF_8);
+
         // Preceding isEmpty() check is for reducing tiny overhead applied for each contains() by calling
         // Object#hashCode. Since ignoreKeys should be empty for most cases..
-        if (!ignoreKeys.isEmpty() && ignoreKeys.contains(record.key())) {
```

Comment:
Reading this again, I thought we might want to changes ignoreKeys to a `Set<ByteBuffer>` to avoid the charset decoding step (=encode only once when reading blacklist config)

Reply:
This is finally fixed by https://github.com/line/decaton/pull/157/commits/6d3cc6c53ed73db999b8adb96dca99d46df28be1